### PR TITLE
docs: updates examples with grafana/pyroscope, enforces consistent style

### PR DESCRIPTION
**Pull request by petethepig**
See commit log for list of changes.

* For most examples I changed `pyroscope/pyroscope` to `grafana/pyroscope`
* For some I had to add Grafana Agent
* Some I deleted

### CODE REVIEW

File: examples/nodejs/express-pull/docker-compose.yml [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+      - '4040:4040'
+
+  agent:
+    image: grafana/agent:latest
+    volumes:
+      - ./agent.config.river:/etc/agent-config/config.river
+    command:
+      - run
+      - /etc/agent-config/config.river
+      - --server.http.listen-addr=0.0.0.0:12345
```
Comment (bryanhuhta) [inline examples/nodejs/express-pull/docker-compose.yml:15]: Is port 12345 correct?

File: examples/ruby/rideshare_rails/config/secrets.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+production:
+  secret_key_base: placeholder
```
Comment (bryanhuhta) [inline examples/ruby/rideshare_rails/config/secrets.yml:2]: Is this actually needed for the example Rails app?

**Review by bryanhuhta — APPROVED** [review]
Few questions, but otherwise LGTM

File: examples/ruby/rideshare_rails/config/secrets.yml (hunk 2 above)
Comment (petethepig) [reply examples/ruby/rideshare_rails/config/secrets.yml:2]: Yeah, I think so

File: examples/nodejs/express-pull/docker-compose.yml (hunk 1 above)
Comment (petethepig) [reply examples/nodejs/express-pull/docker-compose.yml:15]: Yeah

**Pull request merged**